test/scenarios/managed-network/output/src/managed-network/azext_managed_network/tests/latest/preparers.py

```diff
@@ -8,152 +8,72 @@
-    def __init__(self, name_prefix='clitest.nic',
-                 parameter_name='subnet',
-                 resource_group_key=KEY_RESOURCE_GROUP,
-                 vnet_key=KEY_VIRTUAL_NETWORK,
-                 dev_setting_name='AZURE_CLI_TEST_DEV_VNET_NIC_NAME',
-                 key=KEY_VNET_NIC):
-        if ' ' in name_prefix:
-            raise CliTestError(
-                'Error: Space character in name prefix \'%s\'' % name_prefix)
-        super(VnetNicPreparer, self).__init__(name_prefix, 15)
-        self.cli_ctx = get_dummy_cli()
-        self.parameter_name = parameter_name
-        self.key = key
-        self.resource_group = [resource_group_key, None]
-        self.vnet = [vnet_key, None]
-        self.dev_setting_name = os.environ.get(dev_setting_name, None)
+        self.virtual_network_key = virtual_network_key
+        self.resource_group_key = resource_group_key
+        self.name_prefix = name_prefix
+        self.random_name_length = random_name_length
 
     def create_resource(self, name, **_):
-        if self.dev_setting_name:
-            return {self.parameter_name: self.dev_setting_name, }
-
-        if not self.resource_group[1]:
-            self.resource_group[1] = self.test_class_instance.kwargs.get(
-                self.resource_group[0])
-            if not self.resource_group[1]:
-                raise CliTestError("Error: No resource group configured!")
-        if not self.vnet[1]:
-            self.vnet[1] = self.test_class_instance.kwargs.get(self.vnet[0])
-            if not self.vnet[1]:
-                raise CliTestError("Error: No vnet configured!")
-
-        template = 'az network nic create --resource-group {} --name {} --vnet-name {} --subnet default '
-        self.live_only_execute(self.cli_ctx, template.format(
-            self.resource_group[1], name, self.vnet[1]))
-
+        cmd = 'az network vnet subnet create -n {} --vnet-name {} -g {} --address-prefixes "10.0.0.0/21"'
+        cmd = cmd.format(
+            name,
+            self.test_class_instance.kwargs.get(self.virtual_network_key),
+            self.test_class_instance.kwargs.get(self.resource_group_key),
+        )
+        self.live_only_execute(self.cli_ctx, cmd)
         self.test_class_instance.kwargs[self.key] = name
-        return {self.parameter_name: name}
+        return {self.key: name}
 
     def remove_resource(self, name, **_):
-        if not self.dev_setting_name:
-            self.live_only_execute(
-                self.cli_ctx,
-                'az network nic delete --name {} --resource-group {}'.format(name, self.resource_group[1]))
+        cmd = 'az network vnet subnet delete --name {} --resource-group {} --vnet-name {}'
+        cmd = cmd.format(
+            name,
+            self.test_class_instance.kwargs.get(self.resource_group_key),
+            self.test_class_instance.kwargs.get(self.virtual_network_key),
+        )
+        self.live_only_execute(self.cli_ctx, cmd)
```

Comment:
Are the VnetSubnetPreparer and VnetNicPreparer being removed in managed-network expected?

Reply:
is expected. since no Nic is used in managed-network.

This PR create preparers only if they are used in the running RP.